### ISSUE TITLE
Track future updates to Module.extensions

### ISSUE THREAD

**Issue by sidorares** [sidorares/hot-module-replacement#6]
this module monkey-patches known extensions, but for all future extensions the hooks are not added automatically. Proposed solution: instead plain js object with extensions as keys use Proxy and track writing new keys

**Comment by kireerik** [comment]
Custom extensions are also not supported currently. Example: https://stackoverflow.com/questions/12752622/require-file-as-string/12753026#12753026

**Comment by sidorares** [comment]
@kireerik this example works for me:

```js
var fs = require('fs');

require.extensions['.txt'] = function(module, filename) {
  module.exports = fs.readFileSync(filename, 'utf8');
};

require('hot-module-replacement')({});

setInterval(() => {
  var words = require('./words.txt');
  console.log(words);
}, 10000);

if (module.hot) {
  module.hot.accept('./words.txt', () => {
    var words = require('./words.txt');
    console.log('Updated');
    console.log(words);
  });
}
```

note that as I mentioned in this issue extensions must be set up before call to `require('hot-module-replacement')({});` otherwise it's not wired up properly ( this is possible to fix, for example by replacing _extensions object with `Proxy` )

**Comment by kireerik** [comment]
Yes, you are correct. It is only not working if you define a custom extension in a required module for example.
